tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java

```diff
@@ -2228,6 +2244,257 @@ private void setPartitionKeys(int pid, PartitionDescProto.Builder partitionDesc)
     return partitions;
   }
 
+  /**
+   * Check if list of partitions exist on catalog.
+   *
+   *
+   * @param databaseId
+   * @param tableId
+   * @return
+   */
+  public boolean existPartitionsOnCatalog(int tableId) {
+    Connection conn = null;
+    ResultSet res = null;
+    PreparedStatement pstmt = null;
+    boolean result = false;
+
+    try {
+      String sql = "SELECT COUNT(*) CNT FROM "
+        + TB_PARTTIONS +" WHERE " + COL_TABLES_PK + " = ?  ";
+
+      if (LOG.isDebugEnabled()) {
+        LOG.debug(sql);
+      }
+
+      conn = getConnection();
+      pstmt = conn.prepareStatement(sql);
+      pstmt.setInt(1, tableId);
+      res = pstmt.executeQuery();
+
+      if (res.next()) {
+        if (res.getInt("CNT") > 0) {
+          result = true;
+        }
+      }
+    } catch (SQLException se) {
+      throw new TajoInternalError(se);
+    } finally {
+      CatalogUtil.closeQuietly(pstmt, res);
+    }
+    return result;
+  }
+
+  @Override
+  public List<PartitionDescProto> getPartitionsByFilter(PartitionsByFilterProto request) {
+    throw new TajoRuntimeException(new UnsupportedException());
```

Comment:
```UnsupportedException``` means that we don't support this feature. But, this is not true because we support partition pruning with simple filters as well as more complex conditions.
So, it would be good if this method just call ```getPartitionsByAlgebra()``` in the below.